### ISSUE TITLE
Compilation warning in projects that don't have Oban installed

### ISSUE THREAD

**Issue by AndrewDryga** [getsentry/sentry-elixir#752]
Created a new projects with Plug, Phoenix, LiveView and Sentry:
```
=> sentry
Compiling 34 files (.ex)
    warning: Oban.Worker.from_string/1 is undefined (module Oban.Worker is not available or is yet to be defined)
    │
 31 │       case {oban_worker_mod.from_string(job.worker), stacktrace} do
    │                             ~
    │
    └─ lib/sentry/integrations/oban/error_reporter.ex:31:29: Sentry.Integrations.Oban.ErrorReporter.handle_event/4
```

**Comment by whatyouhide** [comment]
This happens in Elixir 1.17, which introduced the start of the type system, which we can't fool that easily anymore 😄 We need to use `apply/3` here. @savhappy wanna work on this?

**Comment by savhappy** [comment]
@whatyouhide on it!

**Comment by savhappy** [comment]
[PR out for review ](https://github.com/getsentry/sentry-elixir/pull/754) @whatyouhide 

Thank you @AndrewDryga for pointing this out (:

**Comment by whatyouhide** [comment]
Closed by #754, thanks @savhappy and thanks @AndrewDryga for reporting 💟
